lib/ConnectionPool.cc

```diff
@@ -34,13 +34,13 @@ DECLARE_LOG_OBJECT()
 namespace pulsar {
 
 ConnectionPool::ConnectionPool(const ClientConfiguration& conf, ExecutorServiceProviderPtr executorProvider,
-                               const AuthenticationPtr& authentication, bool poolConnections,
-                               const std::string& clientVersion)
+                               const AuthenticationPtr& authentication, const std::string& clientVersion)
     : clientConfiguration_(conf),
       executorProvider_(executorProvider),
       authentication_(authentication),
-      poolConnections_(poolConnections),
-      clientVersion_(clientVersion) {}
+      clientVersion_(clientVersion),
+      randomDistribution_(0, conf.getConnectionsPerBroker()),
```

Comment:
```suggestion
      randomDistribution_(0, conf.getConnectionsPerBroker() -1),
```

The range is left-closed and right-closed. We need to `-1` here to make sure the total number of the connections to be equal to `conf.getConnectionsPerBroker()`.

Reply:
Yeah this comment is right.